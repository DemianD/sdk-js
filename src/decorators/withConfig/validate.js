import { createConfigError } from '../../util/createError'

import {
  MISCONFIGURED,
  MISCONFIGURED_AUTH_PROVIDER,
  MISCONFIGURED_LOCALE,
} from '../../constants'

export default function validate(config = {}) {
  if (!(config instanceof Object)) {
    throw createConfigError(
      `expected object but received '${config}'`,
      MISCONFIGURED
    )
  }

  if (!config.hasOwnProperty('locale') || config.locale === undefined) {
    throw createConfigError(
      'locale must be configured.',
      MISCONFIGURED,
      MISCONFIGURED_LOCALE
    )
  }

  /**
   * MANDATORY - Start with 2 a-z values. E.g.: nl
   *
   * OPTIONAL:
   *  - Then have a dash. E.g.: -
   *  - Then have 2 a-z values. E.g.: be
   *  - Then have a space or dash. E.g.:  -
   *  - Then have 1, 2 or 3 a-z or numbers 0-9. E.g.: vwv
   *
   * Resulting in: `nl-be-vwv`, `nl-be` or `nl` (case insensitive)
   */
  const locales = Array.isArray(config.locale) ? config.locale : [config.locale]
  if (
    locales.some(
      locale =>
        !/^[a-z]{2}(?:-[A-Z]{2}(?:[- ]{1}[A-Z09]{1,3})?)?$/i.test(locale)
    )
  ) {
    throw createConfigError(
      'locale is not valid.',
      MISCONFIGURED,
      MISCONFIGURED_LOCALE
    )
  }

  if (
    !config.hasOwnProperty('auth_provider') ||
    config.auth_provider === undefined
  ) {
    throw createConfigError(
      'auth_provider must be configured.',
      MISCONFIGURED,
      MISCONFIGURED_AUTH_PROVIDER
    )
  }

  return config
}
