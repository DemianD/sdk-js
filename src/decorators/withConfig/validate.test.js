import validate from './validate'

describe('config # validate', () => {
  it('should error when null is given', () => {
    expect(() => validate(null)).toThrowErrorMatchingSnapshot()
  })
})
