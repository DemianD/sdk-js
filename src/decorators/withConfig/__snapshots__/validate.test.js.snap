// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config # validate should error when null is given 1`] = `"[CONFIG]: expected object but received 'null'"`;
